analyses/mutational-signatures/04-analyze_de_novo.Rmd

```diff
@@ -0,0 +1,175 @@
+---
+title: "De novo mutational signature extraction from WGS data"
+author: "S. Spielman"
+date: "2021"
+output:
+  html_notebook:
+    toc: yes
+    toc_float: yes
+  html_document:
+    toc: yes
+    df_print: paged
+editor_options:
+  chunk_output_type: inline
+---
+
+
+<br><br><br><br>
+
+#### Packages and paths
+
+
+```{r}
+# Load libraries
+library(sigfit)
+`%>%` <- dplyr::`%>%`
+
+# Path to input data
+path_to_input_rds <- file.path("..", "..", "scratch", "mutational-signatures", "extraction")
+
+# Result path
+path_to_results <- "results/"
+if (!dir.exists(path_to_results)) {
+  dir.create(path_to_results, recursive = TRUE)
+}
+
+# De novo signatures and exposures list (condensed from sigfit output) file
+de_novo_signatures_file <- file.path(path_to_results, "de_novo_signatures.RDS")
+de_novo_exposures_file <- file.path(path_to_results, "de_novo_exposures.RDS")
+
+# Load cosmic data from sigfit
+data("cosmic_signatures_v3")
+cosmic_names <- row.names(cosmic_signatures_v3)
+```
+
+<br><br>
+First, we need to extract the signatures and exposures from the *de novo* extraction RDS files. We save each into a list `de_novo_signatures` and `de_novo_extractions` separately, and write each list to file. If _both_ files already exist, we can simply read it in the lists.
+
+```{r parse_signatures}
+if (file.exists(de_novo_signatures_file) & file.exists(de_novo_exposures_file))
```

Comment:
I'm not sure what my opinion on this suggetion is, but I'll say it anyway - do we want to do something with `params` around overwriting these files? I can imagine a situation where I'd want to do that, but probably I'd think to delete the RDS copies I have?

Reply:
Yeah, this is a good point in case something changes and we _need_ to overwrite. I put in the `if` to preserve some personal sanity since it can take a decent amount of time (15-20 min?) to parse out, but in the scheme of things maybe that's fine for it to take 20 minutes?

Reply:
I don't think 20 min is that long, personally!

Reply:
Mandatory coffee break!